base_mixin_restrict_field_access/models/restrict_field_access_mixin.py

```diff
@@ -72,6 +72,61 @@ def read(self, fields=None, load='_classic_read'):
                         self._fields[field].null(self.env))
         return result
 
+    def read_group(self, cr, uid, domain, fields, groupby, offset=0, limit=None, context=None, orderby=False, lazy=True):
+        """
+        Remove inaccessible fields from 'fields', 'groupby' and 'orderby'.
+
+        If this removes all 'fields', return no records.
+        If this removes all 'groupby', group by first remaining field.
+        If this removes 'orderby', don't specify order.
+        """
+        sanitised_fields = [f for f in fields if self._restrict_field_access_is_field_accessible(cr, uid, [], f)]
```

Comment:
last thing here: To be symmetric to `read`, the result should contain keys for those fields, but with the null value of the field type. This is important in order not to break views, because a lot of code in the web client expects a key for a field to exist if it was requested